security-build-pipelines.yml

```diff
@@ -0,0 +1,170 @@
+        buildType: 'current'
+        downloadType: 'single'
+        downloadPath: '$(Agent.TempDirectory)/artifacts'
+        artifactName: 'deploy'
+
+    - task: DownloadBuildArtifacts@0
+      displayName: 'Download NuGet Packages'
+      inputs:
+        buildType: 'current'
+        downloadType: 'single'
+        downloadPath: '$(Build.SourcesDirectory)/artifacts'
+        artifactName: 'nuget'
+      
+    - task: ExtractFiles@1
+      displayName: 'Extract Stu3 Web Server Binaries'
+      inputs:
+        archiveFilePatterns: '$(Agent.TempDirectory)/artifacts/deploy/Microsoft.Health.Fhir.Stu3.Web.zip'
+        destinationFolder: '$(Build.SourcesDirectory)/artifacts/web/Stu3'
+    - task: ExtractFiles@1
+      displayName: 'Extract R4 Web Server Binaries'
+      inputs:
+        archiveFilePatterns: '$(Agent.TempDirectory)/artifacts/deploy/Microsoft.Health.Fhir.R4.Web.zip'
+        destinationFolder: '$(Build.SourcesDirectory)/artifacts/web/r4'
+    - task: ExtractFiles@1
+      displayName: 'Extract R4B Web Server Binaries'
+      inputs:
+        archiveFilePatterns: '$(Agent.TempDirectory)/artifacts/deploy/Microsoft.Health.Fhir.R4B.Web.zip'
+        destinationFolder: '$(Build.SourcesDirectory)/artifacts/web/r4b'
+    - task: ExtractFiles@1
+      displayName: 'Extract R5 Web Server Binaries'
+      inputs:
+        archiveFilePatterns: '$(Agent.TempDirectory)/artifacts/deploy/Microsoft.Health.Fhir.R5.Web.zip'
+        destinationFolder: '$(Build.SourcesDirectory)/artifacts/web/r5'
+    - task: AntiMalware@4
+      inputs:
+        InputType: 'Basic'
+        ScanType: 'CustomScan'
+        FileDirPath: '$(Build.SourcesDirectory)'
+        EnableServices: true
+        TreatSignatureUpdateFailureAs: 'Standard'
+        SignatureFreshness: 'OneDay'
+        TreatStaleSignatureAs: 'Error'
+
+    - task: Armory@2
+      inputs:
+        targetDirectory: '$(Build.SourcesDirectory)\samples\templates'
+        targetFiles: 'f|*.json'
+        excludePassesFromLog: false
+
+    - task: CredScan@3
+      inputs:
+        scanFolder: ${{ parameters.credScanDirectory }}
+        outputFormat: 'sarif'
+        suppressionsFile: 'CredScanSuppressions.json'
+        verboseOutput: true
+
+    - task: ComponentGovernanceComponentDetection@0
+      displayName: Component Detection
+      inputs:
+        scanType: 'Register'
+        verbosity: 'Verbose'
+        alertWarningLevel: 'High'
+
+    - task: CSRF@1
+      inputs:
+          Path: '$(Build.SourcesDirectory)'
+          ToolVersion: Latest
+          
+    - task: RoslynAnalyzers@3
+      inputs:
+        userProvideBuildInfo: 'autoMsBuildInfo'
+        setupCommandlinePicker: 'vs2022'
+        rulesetName: 'Multiple'
+        multipleRulesetRequired: true
+        multipleRulesetRecommended: true
+        customRuleset: 'CustomAnalysisRules.ruleset'
+        suppressionFileForCompilerWarnings: 'RoslynSuppressions.txt'
+      env:
+        SYSTEM_ACCESSTOKEN: $(System.AccessToken)
```

Comment:
Should the missing steps just be added into: https://github.com/microsoft/fhir-server/blob/main/build/jobs/analyze.yml?

This template already runs in PR and CI pipelines

Reply:
Right that's where I started from https://github.com/microsoft/fhir-server/pull/3165/files
Trying to see if we can directly assign the full path to build.exe or if we chose to do autoMsBuildInfo then we will have to run the build again